damnit/ctxsupport/ctxrunner.py

```diff
@@ -621,12 +621,18 @@ def save_hdf5(self, hdf5_path, reduced_only=False):
                     dataarray = _set_encoding(dataarray)
                 obj = obj.rename_vars(vars_names)
 
+            # HDF5 implements some features that are not yet supported by netcdf,
+            # e.g. data types support (bool, complex, etc.). We don't really care
+            # about netcdf compatibility since we offer an API to access the
+            # data, so we use the `invalid_netcdf` option to be able to use these
+            # features.
             obj.to_netcdf(
                 hdf5_path,
                 mode="a",
                 format="NETCDF4",
                 group=name,
                 engine="h5netcdf",
+                invalid_netcdf=True,
```

Comment:
The h5netcdf docs [only list booleans & references](https://h5netcdf.org/#invalid-netcdf-files) as invalid types, and in my brief test it seemed happy to save complex numbers without this flag.

In the absence of a native compound type in HDF5, h5py stores it as a compound type with fields `r` & `i`, and recognises this when reading data back. That may well mean that other NetCDF libraries read it as a compound (struct) type.

Reply:
what's your environment? In the current `amore` environment I get a:

```python
CompatibilityError: complex dtypes are not a supported NetCDF feature, and are not allowed by h5netcdf unless invalid_netcdf=True.
```

Reply:
thanks, I updated the minimum requirement for h5netcdf